net/portmapper/igd_test.go

```diff
@@ -38,16 +44,19 @@ type igdCounters struct {
 	numPMPDiscoRecv      int32
 	numPCPRecv           int32
 	numPCPDiscoRecv      int32
+	numPCPMapRecv        int32
+	numPCPOtherRecv      int32
 	numPMPPublicAddrRecv int32
 	numPMPBogusRecv      int32
 }
 
-func NewTestIGD() (*TestIGD, error) {
+func NewTestIGD(t *testing.T) (*TestIGD, error) {
```

Comment:
Is t only used for logging? Maybe it should be a `logf logger.Logf` instead then? Then people can pass in `logger.Discard` to shut it up.

Reply:
I added some points where it errors in the tester if it fails to write, but that should be an error on both ends.